ocs_ci/helpers/helpers.py

```diff
@@ -704,6 +704,40 @@ def verify_block_pool_exists(pool_name):
         return False
 
 
+def get_pool_cr(pool_name):
+    """
+    Get the pool cr even kind is unknown
+
+    Args:
+         pool_name (str): The name of the pool to get the cr.
+
+    Returns:
+        dict: If the resource is found.
+        None: If the resource is not found
+
+    """
+    logger.info(f"Checking if pool {pool_name} is kind of {constants.CEPHBLOCKPOOL}")
+    ocp_kind_cephblockpool = ocp.OCP(
+        kind=constants.CEPHBLOCKPOOL, namespace=config.ENV_DATA["cluster_namespace"]
+    )
+    pool_cr = ocp_kind_cephblockpool.get(resource_name=pool_name, dont_raise=True)
+    if pool_cr is not None:
+        return pool_cr
+    else:
+        logger.info(
+            f"Pool {pool_name} is not kind={constants.CEPHBLOCKPOOL}, checkging if it is kind={constants.CEPHFILESYSTEM}"
+        )
+        ocp_kind_cephfilesystem = ocp.OCP(
+            kind=constants.CEPHFILESYSTEM,
+            namespace=config.ENV_DATA["cluster_namespace"],
+        )
+        pool_cr = ocp_kind_cephfilesystem.get(resource_name=pool_name, dont_raise=True)
+        if pool_cr is not None:
```

Comment:
You can return None immediately as this function returns None in case the resource wasn't found

Reply:
Great idea